src/dict.c

```diff
@@ -210,7 +210,7 @@ int dictTryExpand(dict *d, unsigned long size) {
  * work it does would be unbound and the function may block for a long time. */
 int dictRehash(dict *d, int n) {
     int empty_visits = n*10; /* Max number of empty buckets to visit. */
-    if (!dictIsRehashing(d)) return 0;
+    if (!dict_can_resize || !dictIsRehashing(d)) return 0;
```

Comment:
I think this will invalidate the "dict_force_resize_ratio" functionality that occurs on the parent process, since the resize will still occur but the elements won't be rehashed into the new bucket.

Reply:
Soft preference for 3. It seems like the override code was added for a reason and I don't like breaking abstractions.